src/main/src/main/java/org/geoserver/catalog/CoverageViewReader.java

```diff
@@ -378,14 +466,67 @@ public static GridCoverage2DReader wrap(GridCoverage2DReader reader, CoverageVie
             return new StructuredCoverageViewReader((StructuredGridCoverage2DReader) reader,
                     coverageView, coverageInfo, hints);
         } else {
-            return new CoverageViewReader((GridCoverage2DReader) reader, coverageView,
+            return new CoverageViewReader(reader, coverageView,
                     coverageInfo, hints);
         }
     }
 
     @Override
     public Format getFormat() {
-       return delegate.getFormat();
+        return new Format(){
+
+            private final Format delegateFormat = delegate.getFormat();
+        	
+            @Override
+            public ParameterValueGroup getWriteParameters() {
+                return delegateFormat.getWriteParameters();
+            }
+            
+            @Override
+            public String getVersion() {
+                return delegateFormat.getVersion();
+            }
+            
+            @Override
+            public String getVendor() {
+                return delegateFormat.getVendor();
+            }
+            
+            @Override
+            public ParameterValueGroup getReadParameters() {
+                HashMap<String, String> info = new HashMap<String, String>();
+
+                info.put("name", getName());
+                info.put("description", getDescription());
+                info.put("vendor", getVendor());
+                info.put("docURL", getDocURL());
+                info.put("version", getVersion());
+                
+                // reading parameters
+                return new ParameterGroup(
```

Comment:
This read actually delegate reads to a wrapped format, so it should expose all wrapped read params, plus the band one, not just the band one.